keps/sig-windows/4885-windows-cpu-and-memory-affinity/README.md

```diff
@@ -0,0 +1,635 @@
+    // CPU mask relative to this CPU group.
+    uint64 cpu_mask = 1;
+    // CPU group that this CPU belongs to.
+    uint32 cpu_group = 2;
+}
+```
+
+Since the Kubelet API's are looking for a distinct ProcessorId, the processorid's will be calculated by looping 
+through the mask and calculating the ids with `(group *64) + procesorid` resulting in unique processor id's from `group 0` as `0-63` and 
+processor Id's from `group 1` as `64-127` and so on. This translation will be done only in kubelet, the `cpu_mask` will be used when 
+communicating with the container runtime.
+
+```golang
+for i := 0; i < 64; i++ {
+		if groupaffinity.Mask&(1<<i) != 0 {
+			processors = append(processors, i+(int(a.Group)*64))
+		}
+	}
+}
+```
+
+Using this logic, a cpu bit mask of `0000111` (leading zero's removed) would result in cpu's: 
+
+- `0,1,2` in `group 0` 
+- `64,65,66` in `group 1`.
+
+When converting back to the Windows Group Affinity we will divide the cpu number by 64 to get the group number then 
+use mod of 64 to calculate the location of the cpu in mask:
+
+```golang
+group := cpu / 64
+mask := 1 << (cpu % 64)
+groupaffinity.Mask |= mask
+```
+
+There are some scenarios where cpu count might be greater than 64 cores but in each group it is less
+than 64. For instance, you could have 2 CPU groups with 35 processors each.  The unique ID's using the strategy 
+above would give you: 
+
+- CPU group 0 : 0 to 34
+- CPU group 2: 64 to 99
+
+### Windows Memory considerations
+
+[Numa nodes](https://learn.microsoft.com/en-us/windows/win32/procthread/numa-support) can not be directly assigned or guaranteed via the Windows API but the windows sub system attempts to use memory assigned to the CPU to improve performance.  
+It is possible to indicate to a process which Numa node is preferred but a limitation of the Windows API's is that [PROC_THREAD_ATTRIBUTE_PREFERRED_NODE](https://learn.microsoft.com/windows/win32/api/processthreadsapi/nf-processthreadsapi-updateprocthreadattribute)
+does not support setting multiple Numa nodes for a single Job object (i.e. Container) so is not usable in the context of Windows containers which have multiple processes.  
+
+Since the existing Memory Manager Policy `Static` on Linux has semantic meaning that ensures that only the memory from a 
+NUMA node selected is used. We can not re-use this policy on Windows given that there is no way to ensure only the memory 
+on the Node that the memory manager selects.  For these reason if the `Static` policy is chosen on Windows kubelet will fail
+to start with an error message that states it can use the `Static` policy.  Instead we will create a new Windows only Policy called
+`BestEffort` which will initially only be implemented on Windows and Linux will fail to start if the Policy is set.  
+We do not have any use cases for this policy to be implemented on Linux at this time and so we will avoid adding a feature that isn't 
+applicable to that platform.
```

Comment:
after our conversations I'm fine also supporting the policy on Linux as stretch goal. IOW, I'm fine implementing `BestEffort` only on windows.